WordPress/Classes/ViewRelated/Menus/Views/MenuItemTypeViewController.h

```diff
@@ -0,0 +1,46 @@
+#import <UIKit/UIKit.h>
+#import "MenuItem.h"
+
+@class Blog;
+
+@protocol MenuItemTypeSelectionViewDelegate;
+
+@interface MenuItemTypeViewController : UIViewController
+
+@property (nonatomic, weak) id <MenuItemTypeSelectionViewDelegate> delegate;
+
+/**
+ The itemType to display as selected in the UI, such as MenuItemTypePage.
+ */
+@property (nonatomic, strong) NSString *selectedItemType;
+
+/**
+ Fetch the postTypes available for the blog, including custom post types.
+ */
+- (void)loadPostTypesForBlog:(Blog *)blog;
+
+/**
+ Helper method for updating the layout if the parentView's layout changed.
+ */
+- (void)updateDesignForLayoutChangeIfNeeded;
+
+/**
+ Ensure the selected itemType is viisble on screen.
+ */
+- (void)focusSelectedTypeViewIfNeeded:(BOOL)animated;
+
+@end
+
+@protocol MenuItemTypeSelectionViewDelegate <NSObject>
```

Comment:
At first I thought `MenuItemTypeSelectionViewDelegate` should be moved to `MenuItemTypeSelectionView` but its actually a controller delegate isn't it. Maybe we could rename it for clarity?